### ISSUE TITLE
Mode / Lambda Functions in Groupby Agg Arguments

### ISSUE THREAD

**Issue by samuelroberth** [dask/dask#2632]
I see in the source code that only count / size / mean / sum / min / max are currently supported by agg for groupbys. Is there any plans to implement mode / allow lambda functions or is there a hacky way to do this? Specifically I want to get the most common string for a given column (like passing lambda x:x.value_counts().index[0]).

Thanks.

**Comment by mrocklin** [comment]
See https://github.com/dask/dask/pull/2344

If you wanted to take that over and try to get it in that would be most welcome!

**Comment by samuelroberth** [comment]
I'll have a look.

**Comment by samuelroberth** [comment]
Any idea how I can start to approach this from a relatively newbie perspective? I'm pretty lost in the source code at the moment.

**Comment by samuelroberth** [comment]
Are custom reductions for groupby available or is that not coded?

**Comment by TomAugspurger** [comment]
@samuelroberth currently just the aggregations implemented in dask work. #2344 is providing a way for any custom aggregation function to work.

**Comment by samuelroberth** [comment]
Ok thanks Tom. I didn't see that code in the source so was just wondering. I'm really trying to get a 'mode' implemented but it is proving difficult. I am trying to adapt from the variance but the logic is not really transferable.

**Comment by TomAugspurger** [comment]
Gotcha.

One issue with `mode` is that (in pandas) it's not always an aggregation:

```python
In [10]: s = pd.Series([1, 1, 2, 2, 3])

In [11]: s.mode()
Out[11]:
0    1
1    2
dtype: int64
```

That's why `df.groupby(grouper).agg(pd.Series.mode)` fails in pandas, and typically dask sticks to pandas' semantics. 

I've pinged them in the other issue, we'll see if they are able to finish up that PR.

**Comment by samuelroberth** [comment]
So once the PR is finished with Pandas than Dask will follow with that? I'm under the impression that Pandas already has custom aggregations (see: https://medium.com/@IceS2/writing-custom-aggregation-functions-with-pandas-96f5268a8596)

**Comment by samuelroberth** [comment]
I think this can be closed as per #2344.
